Makefile

```diff
@@ -326,20 +336,28 @@ test-single-gtest-case: ## Run a single GoogleTest test case: `make test-single-
 		--gtest_filter=*${TEST_CASE_FILTER}* --gtest_repeat=${NUM_REPEATS__} $${break_on_failure_opt}";
 
 .PHONY: clean
-clean: ## Clean Concord-BFT build directory
+clean: create-ccache-folder ## Clean Concord-BFT build directory
 	docker run ${BASIC_RUN_PARAMS} \
 		${CONCORD_BFT_CONTAINER_SHELL} -c \
 		"rm -rf ${CONCORD_BFT_BUILD_DIR}"
 
 .PHONY: clean-all
-clean-all: ## Clean Concord-BFT build directory and any other untracked/ignored files.For a 'dry run' use DRY_RUN=TRUE, for an interactive run use INTER=TRUE.
+clean-all: ## Clean Concord-BFT build directory and any other untracked/ignored files. Does not include ccache cache folder.For a 'dry run' use DRY_RUN=TRUE, for an interactive run use INTER=TRUE.
```

Comment:
Why it does not include the cache dir?
 When we clean, we want a fresh build without any use of cache and reuse of objects no?

Reply:
Well, my idea says there are 3 levels of clean, but I guess there is no single truth here. What makes sense to me might not make sense to others:

- 'make clean' is regular clean of project bin -we dont want to delete ccache since we want to clean local build foder, and keep shared cache with other projects.
- make 'clean-all' - clean build and all repository artifacts. it still cleans things under CURRENT project. ccache is above prohject/folder/concord-bft level, it is a shared (or remote) cache shared by all users (projects, hosts or users)
- make 'clean-ccache-cache' - the only expclit command which will clean the ccache, and for me it makes sensse since cleaning the whole shared cache for everyone shouldn't be on the project level. this one goes to $HOME/.ccache/ by default and delete everything under.

Reply:
Im going to add explentation in confluence about it.
It think my logic make sense here. Other people might find a different solution more appropriate.

Reply:
I think all the many cleans created confusion.
Tere going to be a single make clean which cleans only build folder. Using flags it will be possible to clean different things. For more infor see confluene page or make help.